bindings/wasm/src/credential/revocation/status_list_2021/credential.rs

```diff
@@ -0,0 +1,251 @@
+  /// Returns the created `credentialStatus`.
+  #[wasm_bindgen(js_name = "setCredentialStatus")]
+  pub fn set_credential_status(
+    &mut self,
+    credential: &mut WasmCredential,
+    index: usize,
+    flag: bool,
+  ) -> Result<WasmStatusList2021Entry> {
+    let entry = self
+      .inner
+      .set_credential_status(&mut credential.0, index, flag)
+      .map_err(|e| JsValue::from(JsError::new(&e.to_string())))?
+      .clone();
+    self.wasm_credential = WasmCredential(self.inner.clone().into_inner());
+
+    Ok(WasmStatusList2021Entry(entry))
+  }
+
+  /// Returns the {@link StatusPurpose} of this {@link StatusList2021Credential}.
+  #[wasm_bindgen]
+  pub fn purpose(&self) -> WasmStatusPurpose {
+    self.inner.purpose().into()
+  }
+
+  /// Returns the state of the `index`-th entry, if any.
+  #[wasm_bindgen]
+  pub fn entry(&self, index: usize) -> Result<WasmCredentialStatus> {
+    self
+      .inner
+      .entry(index)
+      .map(WasmCredentialStatus::from)
+      .map_err(|e| JsError::new(&e.to_string()).into())
+  }
+
+  #[wasm_bindgen(js_name = "clone")]
+  pub fn wasm_clone(&self) -> WasmStatusList2021Credential {
+    self.clone()
+  }
+
+  #[wasm_bindgen(js_name = "fromJSON")]
+  pub fn from_json(json: JsValue) -> Result<WasmStatusList2021Credential> {
+    use crate::error::WasmResult;
+    json.into_serde::<WasmStatusList2021Credential>().wasm_result()
+  }
+
+  #[wasm_bindgen(js_name = "toJSON")]
+  pub fn to_json(&self) -> Result<JsValue> {
+    use crate::error::WasmResult;
+    JsValue::from_serde(self).wasm_result()
+  }
+}
+
+/// Builder type to construct valid {@link StatusList2021Credential} istances.
+#[wasm_bindgen(js_name = StatusList2021CredentialBuilder)]
+pub struct WasmStatusList2021CredentialBuilder(StatusList2021CredentialBuilder);
+
+#[wasm_bindgen(js_class = StatusList2021CredentialBuilder)]
+impl WasmStatusList2021CredentialBuilder {
+  /// Creates a new {@link StatusList2021CredentialBuilder}.
+  #[wasm_bindgen(constructor)]
+  pub fn new(status_list: Option<WasmStatusList2021>) -> WasmStatusList2021CredentialBuilder {
+    Self(StatusList2021CredentialBuilder::new(status_list.unwrap_or_default().0))
+  }
+
+  /// Sets the purpose of the {@link StatusList2021Credential} that is being created.
+  #[wasm_bindgen]
+  pub fn purpose(mut self, purpose: WasmStatusPurpose) -> WasmStatusList2021CredentialBuilder {
+    self.0 = self.0.purpose(purpose.into());
+    self
+  }
+
+  /// Sets `credentialSubject.id`.
+  #[wasm_bindgen(js_name = "subjectId")]
+  pub fn subject_id(mut self, id: String) -> Result<WasmStatusList2021CredentialBuilder> {
+    let id = Url::parse(id).wasm_result()?;
+    self.0 = self.0.subject_id(id);
+
+    Ok(self)
+  }
+
+  /// Sets the expiration date of the credential.
+  #[wasm_bindgen(js_name = "expirationDate")]
+  pub fn expiration_date(mut self, time: WasmTimestamp) -> WasmStatusList2021CredentialBuilder {
```

Comment:
Difficult topic, but we hardcode a VC Data Model 1.1 assumption here, could we also hardcode the 2.0 `validFrom`/`validUntil` or would that break in verification later?